### PR TITLE
Feature/advertisments list

### PR DESCRIPTION
**Pull request by tavisca-mrathore**
@achirametavisca @tavisca-smohite @tavisca-rnayyar

### CODE REVIEW

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GetAdvertismentDetailsService {
+
+  private AD_DETAILS_REST_API_SERVER = "https://demo3944353.mockable.io/get-advertisment-details";
```
Comment (achirametavisca) [inline src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: move this to common class

File: src/app/app.component.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
+app-advertisments-list {
```
Comment (achirametavisca) [inline src/app/app.component.css]: Remove this
home page story covers this.

File: src/app/app.component.html [hunk 3]
```diff
@@ -1 +1 @@
-
+<app-advertisments-list></app-advertisments-list>
```
Comment (achirametavisca) [inline src/app/app.component.html]: Remove this
home page story covers this.

File: src/app/home-module/advertisments-list/advertisments-list.component.html [hunk 4]
```diff
@@ -1 +1,11 @@
-<p>advertisments-list works!</p>
+<div *ngFor="let ad of adsList" class="adv-card">
```
Comment (achirametavisca) [inline src/app/home-module/advertisments-list/advertisments-list.component.html]: Add routing to this component and test this component separately.

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  constructor(private httpClient: HttpClient) { }
+
+  public sendGetRequest() {
```
Comment (achirametavisca) [inline src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: Give name that describes what this function is doing.

File: src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class GetAdvertismentsListService {
+
+  private ADS_LIST_REST_API_SERVER = "https://demo4724356.mockable.io/get-advertisments-list";
+
+  constructor(private httpClient: HttpClient) { }
+
+  public sendGetRequest() {
```
Comment (achirametavisca) [inline src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts]: Give name that describes what this function is doing.

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
```
Comment (achirametavisca) [inline src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: **You don't need two different services**

get-advertisments-list.service.ts 
get-advertisment-details.service.ts

Create one service advertismentsService that will have different methods related to advertisments

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts (hunk 5 above)
Comment (achirametavisca) [inline src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: You should mention return type of this function.
This function should return observable<Type>
 ```
/** POST: add a new hero to the database */
addHero (hero: Hero): Observable<Hero> {
  return this.http.post<Hero>(this.heroesUrl, hero, httpOptions)
    .pipe(
      catchError(this.handleError('addHero', hero))
    );
}
```
In component where this function is called, you should handle success, failure cases everytime.
Follow this for all public functions in service
Discuss with me if having any doubt. 
>

File: src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GetAdvertismentsListService {
+
+  private ADS_LIST_REST_API_SERVER = "https://demo4724356.mockable.io/get-advertisments-list";
```
Comment (achirametavisca) [inline src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts]: move to common class

**Review by achirametavisca — APPROVED** [review]

**Review by achirametavisca — CHANGES_REQUESTED** [review]
Check review comments

File: src/app/app.component.css (hunk 2 above)
Comment (tavisca-mrathore) [reply src/app/app.component.css]: Done

File: src/app/app.component.html (hunk 3 above)
Comment (tavisca-mrathore) [reply src/app/app.component.html]: Done

File: src/app/home-module/advertisments-list/advertisments-list.component.html (hunk 4 above)
Comment (tavisca-mrathore) [reply src/app/home-module/advertisments-list/advertisments-list.component.html]: Done

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts (hunk 5 above)
Comment (tavisca-mrathore) [reply src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: Done

File: src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts (hunk 6 above)
Comment (tavisca-mrathore) [reply src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts]: Done

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts (hunk 7 above)
Comment (tavisca-mrathore) [reply src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: Done

File: src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts (hunk 8 above)
Comment (tavisca-mrathore) [reply src/app/services/GetAdvertismentsList/get-advertisments-list.service.ts]: Done

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts (hunk 1 above)
Comment (tavisca-mrathore) [reply src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: Done

File: src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts (hunk 5 above)
Comment (tavisca-mrathore) [reply src/app/services/GetAdvertismentDetails/get-advertisment-details.service.ts]: done

**Review by achirametavisca — APPROVED** [review]
Looks good

**Pull request merged**